homeassistant/components/flexpool/config_flow.py

```diff
@@ -0,0 +1,63 @@
+"""Config flow for Flexpool integration."""
+import logging
+import re
+
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("address"): str,
+        vol.Optional("workers"): bool,
+        vol.Optional("pool"): bool,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    address = data["address"]
+    if not re.findall("^0x[a-fA-F0-9]{40}$", address):
+        raise InvalidAddress
+
+    return {"title": address}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Flexpool."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except InvalidAddress:
+            errors["base"] = "invalid_address"
+        else:
+            return self.async_create_entry(title=info["title"], data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
```

Comment:
Best practice is to start with `if user_input is not None` and handle the checking first. Then after the if block, return the form. Also, not a bad idea to have a catch all exception check here. 
```suggestion
        errors = {}
        if user_input is not None:
          try:
              info = await validate_input(self.hass, user_input)
          except InvalidAddress:
              errors["base"] = "invalid_address"
          except Exception as error:  # pylint: disable=broad-except
              _LOGGER.exception("Unexpected exception: %s", error)
              errors["base"] = "unknown"
          return self.async_create_entry(title=info["title"], data=user_input)

        return self.async_show_form(
            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
        )
```

Reply:
Thank you. 
Though the problem with the catch all is, how would I make the test cover that? It was there before but I removed it because of the test. I personally wouldn't know how to do that.

Reply:
You'll have to patch the method that's being called in your library to return some kind of exception. `HomeAssistantError` is a good one to use. Let me know if you aren't familiar with patches, I can help a little more but I'll have to take a look at the library to see what's happening and what should be patched.

Reply:
Here's an example from the integration I wrote. It patches the `faadelays.Airport.update` method because that is what the library is using during the setup process to retrieve data. https://github.com/home-assistant/core/blob/86d3baa34e4c296f112b4bbc9540f4b0461379aa/tests/components/faa_delays/test_config_flow.py#L105-L120

Reply:
Okay, I think I managed to do it. Gonna test it now. Thank you very much for the example